src/type_translator.ts

```diff
@@ -497,12 +507,13 @@ export class TypeTranslator {
   }
 
   private translateUnion(type: ts.UnionType): string {
-    let parts = type.types.map(t => this.translate(t));
     // Union types that include literals (e.g. boolean, enum) can end up repeating the same Closure
     // type. For example: true | boolean will be translated to boolean | boolean.
     // Remove duplicates to produce types that read better.
-    parts = parts.filter((el, idx) => parts.indexOf(el) === idx);
-    return parts.length === 1 ? parts[0] : `(${parts.join('|')})`;
+    const parts = new Set(type.types.map(t => this.translate(t)));
+    // If it's a single element set, return the single member.
+    if (parts.size === 1) return parts.values().next().value;
+    return `(${Array.from(parts.values()).join('|')})`;
```

Comment:
Does this preserve order?  I guess Sets must behave like other JS structures, or all the tests would become unstable.

Reply:
Yes, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#Description